subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildIntegrationTest.groovy

```diff
@@ -963,6 +963,8 @@ task generate(type: TransformerTask) {
             def srcDir = file('src')
             // Note: task mutates inputs _without_ declaring any inputs or outputs
             task src1 {
+                outputs.dir(srcDir)
+                outputs.upToDateWhen { false }
```

Comment:
This still contradicts with the comment above. Not sure how this should actually work. Exclude the test from partial invalidation?

Reply:
I finally fixed the comment. Let's not exclude the test from partial invalidation.